x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java

```diff
@@ -177,7 +172,7 @@ static IndexLifecycleExplainResponse getIndexLifecycleExplainResponse(
                     lifecycleState.phase(),
                     lifecycleState.action(),
                     // treat a missing policy as if the index is in the error step
-                    indexLifecycleService.policyExists(policyName) == false ? ErrorStep.NAME : lifecycleState.step(),
+                    policyExists == false ? ErrorStep.NAME : lifecycleState.step(),
```

Comment:
The `IndexLifecycleService` only runs on the master node, which meant that `policyExists` was always `false`. I changed it to use the `Metadata` to check for policy existence.